urllib3/response.py

```diff
@@ -182,6 +192,38 @@ def tell(self):
         """
         return self._fp_bytes_read
 
+    def _init_length(self, request_method):
+        """
+        Set initial length value for Response content.
+
+        Logic extracted from httplib.py in the standard Python library.
+        """
+        length = self.headers.get('content-length')
+        if length is not None and not self.chunked:
+            try:
+                length = int(length)
+            except ValueError:
+                length = None
+            else:
+                if length < 0:
+                    length = None
+        else:
+            length = None
+
+        # Convert status to int for comparison
+        # In some cases, httplib returns a status of "_UNKNOWN"
+        try:
+            status = int(self.status)
+        except:
+            status = 0
+
+        # Check for responses that shouldn't include a body
+        if (status in (204, 304) or 100 <= status < 200
+           or request_method == 'HEAD'):
+            length = 0
+
+        return length
```

Comment:
Nevermind, disregard this. The gain probably doesn't outweigh the loss. This would be breaking, it doesn't really provide significant benefit to the user, and will cripple some of the "do anything" functionality choices Kenneth wanted to preserve in Requests.